ui/desktop/src/components/ParameterInputModal.tsx

```diff
@@ -21,8 +21,33 @@ const ParameterInputModal: React.FC<ParameterInputModalProps> = ({
   useEffect(() => {
     const initialValues: Record<string, string> = {};
     parameters.forEach((param) => {
-      if (param.default) {
-        initialValues[param.key] = param.default;
+      // Set default value for optional parameters if they have a default value
+      if (param.requirement === 'optional' && param.default) {
+        // Validate default value based on parameter type
+        let defaultValue = param.default;
```

Comment:
hmm, we set the defaultValue here and then check a bunch of things where if we like the values, we set defaultValue to mostly the same value. 

in general we should probably set the validity of the recipe when we read it, not at this point. but here I think we can just leave out everything and just go
```
if (param.requirement === 'optional' && param.default) {
    initialValues[param.key] = param.default
}
```
possibly lowercasing it if param.input_type == 'boolean'